packages/cosmic-swingset/lib/ag-solo/vats/lib-wallet.js

```diff
@@ -108,20 +123,30 @@ export async function makeWallet(
 
     // Let the caller do what they want with the seat.
     // We'll resolve when deposited.
-    const depositedP = payoutPAP.then(payoutAP =>
-      Promise.all(payoutAP).then(payoutA =>
+    const depositedP = payoutObjP.then(payoutObj => {
+      const payoutIndexToRole = [];
+      return Promise.all(
+        Object.entries(payoutObj).map(([role, payoutP], i) => {
```

Comment:
Why is `Object.entries(payoutObj).map()` better than `payoutA.map()`?

Reply:
Because `payoutObj` may be an object with roles as keys.  The "indexed" version is a degenerate case where `payoutObj` is an array (an object with numeric roles as keys).